src/emulate.c

```diff
@@ -622,7 +611,9 @@ static void block_translate(riscv_t *rv, block_map_t *map, block_t *block)
         for (int j = 1; j < count; j++, next_ir = next_ir->next) {         \
             memcpy(ir->fuse + j, next_ir, sizeof(opcode_fuse_t));          \
         }                                                                  \
-        ir->tailcall = next_ir->tailcall;                                  \
+        for (int j = 1; j < count; j++) {                                  \
+            remove_next_ir(rv, ir, block);                                 \
```

Comment:
Maybe we can just merge this loop with its previous one, which has a similar for-loop condition?